query/result.go

```
@@ -157,10 +175,19 @@ type flusher interface {
 
 func (e *DelimitedMultiResultEncoder) Encode(w io.Writer, results ResultIterator) (int64, error) {
 	wc := &iocounter.Writer{Writer: w}
+
 	for results.More() {
 		result := results.Next()
 		if _, err := e.Encoder.Encode(wc, result); err != nil {
-			return wc.Count(), err
+			// If we have an error that's from
+			// encoding specifically, return it
+			if _, ok := err.(*EncoderError); ok {
+				return wc.Count(), err
+			}
+			// Otherwise, the error is from query execution,
+			// so we encode it instead.
+			e.Encoder.EncodeError(wc, err)
```

We need to return any potential err from encoding the error here.